arm-insights/2015-04-01/swagger/insightsClient.json

```diff
@@ -0,0 +1,1123 @@
+        },
+        "resourceId": {
+          "type": "string",
+          "description": "Gets or sets the resource identifier of the resource that has emitted the log."
+        },
+        "retention": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Gets or sets the retention."
+        },
+        "blobLocation": {
+          "$ref": "#/definitions/BlobLocation",
+          "description": "Gets or sets the location of the blob."
+        }
+      },
+      "description": "Log definition class specifies the metadata for a log."
+    },
+    "LogDefinitionCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LogDefinition"
+          },
+          "description": "Gets or sets the values for the log definitions."
+        }
+      },
+      "description": "Represents collection of log definitions."
+    },
+    "LogValue": {
+      "properties": {
+        "time": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the time when the log was generated"
+        },
+        "resourceId": {
+          "type": "string",
+          "description": "Gets or sets id of the reosurce related to the log."
+        },
+        "operationName": {
+          "type": "string",
+          "description": "Gets or sets the name of the operation."
+        },
+        "operationVersion": {
+          "type": "string",
+          "description": "Gets or sets the version of the operation."
+        },
+        "category": {
+          "type": "string",
+          "description": "Gets or sets the category of the log."
+        },
+        "resultType": {
+          "type": "string",
+          "description": "Gets or sets the status of the operation."
+        },
+        "resultSignature": {
+          "type": "string",
+          "description": "Gets or sets the substatus of the operation."
+        },
+        "resultDescription": {
+          "type": "string",
+          "description": "Gets or sets the substatus of the operation."
+        },
+        "durationMs": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the duration of the operation in milliseconds."
+        },
+        "callerIpAddress": {
+          "type": "string",
+          "description": "Gets or sets the caller IP address."
+        },
+        "correlationId": {
+          "type": "string",
+          "description": "Gets or sets the correlation id of the log.  Used to group together a set of related log."
+        },
+        "identity": {
+          "type": "string",
+          "description": "Gets or sets the identity that generated the event."
+        },
+        "level": {
+          "type": "string",
+          "description": "Gets or sets the level (Informational/Warning/Error/Critical) of the event."
```

Comment:
If the level has a finite set of values then it is recommended to make this as an Enum. Please use the x-ms-enum extension properly in such a scenario. Having an enum makes discoverability easy for users.